### PR TITLE
Fix an ArrayIndexOutOfBoundsException bug

### PR DESCRIPTION
**Pull request by Pr0methean**
Was caused by concurrent access to a non-thread-safe list. Using a ConcurrentLinkedDeque instead.

### CODE REVIEW

File: src/main/java/net/glowstone/entity/GlowPlayer.java [hunk 1]
```diff
@@ -140,6 +142,7 @@
 import org.bukkit.advancement.AdvancementProgress;
 import org.bukkit.block.Block;
 import org.bukkit.block.BlockFace;
+import org.bukkit.block.BlockState;
```
Comment (aramperes) [inline src/main/java/net/glowstone/entity/GlowPlayer.java]: Unused import

File: src/main/java/net/glowstone/entity/GlowPlayer.java [hunk 2]
```diff
@@ -901,39 +904,37 @@ protected void jump() {
...
-                        Map<BlockVector, BlockChangeMessage> map = chunks
-                                .computeIfAbsent(key, k -> new HashMap<>());
-                        map.put(new BlockVector(message.getX(), message.getY(), message
-                                .getZ()), message);
-                    }
-                }
+        // separate messages by chunk
+        // inner map is used to only send one entry for same coordinates
+        Map<Key, Map<BlockVector, BlockChangeMessage>> chunks = new HashMap<>();
+        while (true) {
```
Comment (aramperes) [inline src/main/java/net/glowstone/entity/GlowPlayer.java:909]: `while (!blockChanges.isEmpty())` ?

Removes the need for the null check afterwards.

**Review by aramperes — DISMISSED** [review]

File: src/main/java/net/glowstone/entity/GlowPlayer.java (hunk 2 above)
Comment (Pr0methean) [reply src/main/java/net/glowstone/entity/GlowPlayer.java:909]: That creates a race condition if multiple threads call processBlockChanges.

File: src/main/java/net/glowstone/entity/GlowPlayer.java (hunk 1 above)
Comment (Pr0methean) [reply src/main/java/net/glowstone/entity/GlowPlayer.java]: Removed in 73dbfbe

**Review by mastercoms — APPROVED** [review]

**Review by aramperes — APPROVED** [review]

**Pull request merged**